### ISSUE TITLE
Gemspec permissions issue

### ISSUE THREAD

**Issue by SteveEdson** [caskroom/homebrew-cask#13015]
I've started receiving this issue:

```
$ brew cask
Error: Permission denied - /Library/Ruby/Gems/2.0.0/specifications/bundler-1.10.6.gemspec
```

I'm sure its a system specific thing, but it only happens when running cask.

These are the only warnings I get for `brew doctor`:

```
$ brew doctor

Warning: Unbrewed header files were found in /usr/local/include.
Warning: Homebrew's sbin was not found in your PATH but you have installed
Warning: Some keg-only formula are linked into the Cellar.
```

Any help would be appreciated. Thanks.

```
$ brew update && brew upgrade brew-cask && brew cleanup && brew cask cleanup
Already up-to-date.
Error: caskroom/cask/brew-cask 0.56.0 already installed
```

**Comment by vitorgalvao** [comment]
What about `brew cask doctor`?

**Comment by SteveEdson** [comment]
Can't even get that far, just get the gemspec error.

**Comment by vitorgalvao** [comment]
[Have you tried uninstalling and reinstalling homebrew-cask](https://www.youtube.com/watch?v=kb2gzteVNa4)?

**Comment by SteveEdson** [comment]
Ha! Hmm, it didn't work, however it still had the issue while it was uninstalled??

```
$ brew uninstall --force brew-cask
Uninstalling brew-cask... (5528 files, 22M)
```

How would this bit be possible?

```
$ brew cask
Error: Permission denied - /Library/Ruby/Gems/2.0.0/specifications/bundler-1.10.6.gemspec
```

```
$ brew install caskroom/cask/brew-cask
==> Installing brew-cask from caskroom/homebrew-cask
==> Cloning https://github.com/caskroom/homebrew-cask.git
Updating /Library/Caches/Homebrew/brew-cask--git
==> Checking out tag v0.56.0
🍺  /usr/local/Cellar/brew-cask/0.56.0: 2802 files, 11M, built in 18 seconds
```

```
$ brew cask
Error: Permission denied - /Library/Ruby/Gems/2.0.0/specifications/bundler-1.10.6.gemspec
```

**Comment by vitorgalvao** [comment]
Issue is definitely in your setup. What’s the output of `which brew-cask`?

**Comment by vitorgalvao** [comment]
Issue is definitely in your setup. Uninstall homebrew-cask again. After that, what’s the output of `which brew-cask`?

**Comment by SteveEdson** [comment]
I thought it would be, I just can't figure out why.

```
$ brew uninstall brew-cask
Uninstalling /usr/local/Cellar/brew-cask/0.56.0... (2802 files, 11M)
```

```
$ which brew-cask
brew-cask not found
```

```
$ brew cask
Error: Permission denied - /Library/Ruby/Gems/2.0.0/specifications/bundler-1.10.6.gemspec
```

```
$ brew install brew-cask
==> Installing brew-cask from caskroom/homebrew-cask
==> Cloning https://github.com/caskroom/homebrew-cask.git
Updating /Library/Caches/Homebrew/brew-cask--git
==> Checking out tag v0.56.0
🍺  /usr/local/Cellar/brew-cask/0.56.0: 2802 files, 11M, built in 14 seconds

$ which brew-cask
/usr/local/bin/brew-cask
```

**Comment by vitorgalvao** [comment]
```shell
$ which ruby bundler
$ ruby --version
$ bundler --version
$ ls -l /Library/Ruby/Gems/2.0.0/specifications/bundler-1.10.6.gemspec
```

Have you installed any of those (`ruby`, `bundler`, `homebrew`, `homebrew-cask`) as root, at any point.

**Comment by SteveEdson** [comment]
I don't recall doing so, but this is probably where the issue is:

```
$ which ruby bundler
/usr/bin/ruby
bundler not found
```

```
$ ruby --version
ruby 2.0.0p481 (2014-05-08 revision 45883) [universal.x86_64-darwin14]
```

```
$ bundler --version
zsh: permission denied: bundler
```

```
$ ls -l /Library/Ruby/Gems/2.0.0/specifications/bundler-1.10.6.gemspec
-rw-------  1 root  wheel  1902  4 Aug 14:45 /Library/Ruby/Gems/2.0.0/specifications/bundler-1.10.6.gemspec
```

**Comment by vitorgalvao** [comment]
Have you experimented with trashing that problematic file temporarily?

**Comment by SteveEdson** [comment]
The plot thickens...

```
$ sudo rm /Library/Ruby/Gems/2.0.0/specifications/bundler-1.10.6.gemspec
$ brew cask
Error: Permission denied - /Library/Ruby/Gems/2.0.0/specifications/github_api-0.12.4.gemspec
$ sudo rm /Library/Ruby/Gems/2.0.0/specifications/github_api-0.12.4.gemspec
$ brew cask
Error: Permission denied - /Library/Ruby/Gems/2.0.0/specifications/guard-2.13.0.gemspec
```

**Comment by vitorgalvao** [comment]
I said temporarily, as in moving it to another directory just to see if we could get another clue. It seems like you might’ve installed a bunch of gems as root, though, but that doesn’t necessarily mean this error should be occurring.

It’s clear you’re using the system ruby, though. That, together with `sudo` is surely what cause the problem — this is a permissions issue, after all. I searched a bit, and every problem similar to this one seems to have been solved by the asked installing RVM (or similar) and not using the system ruby.
